test/models/UserModel/test.js

```diff
@@ -0,0 +1,322 @@
+      usersCount = _.random(1, 10);
+      for (let iter = 0; iter < usersCount; iter++) {
+        await UsersFactory.Create();
+      }
+      condition = { alias: userAlias };
+    });
+    it('Should return an array', async () => {
+      const { usersData } = await UserModel.find({
+        condition,
+        select: { alias: user.alias },
+        skip: 0,
+      });
+      expect(Array.isArray(usersData)).to.be.true;
+    });
+    it('Should return correct user', async () => {
+      const { usersData } = await UserModel.find({
+        condition,
+        select: { alias: user.alias },
+        skip: 0,
+      });
+      expect(usersData[0].alias).to.be.eq(userAlias);
+    });
+    it('Should return an empty array if user not found', async () => {
+      const wrongName = faker.name.firstName();
+      const { usersData } = await UserModel.find({
+        condition: { name: wrongName },
+        select: { name: wrongName },
+        sort: 'name',
+        skip: 0,
+        limit: 10,
+      });
+      expect(_.isEmpty(usersData)).to.be.true;
+    });
+    it('Should check that the error exists', async () => {
+      const { error } = await UserModel.find({
+        test: _.random(0, 50),
+        host: faker.random.string(),
+        skip: name,
+      });
+      expect(error).to.be.exist;
+    });
+  });
+  describe('On findWithSelect', async () => {
+    let usersCount;
+    beforeEach(async () => {
+      await dropDatabase();
+      usersCount = _.random(1, 10);
+      for (let iter = 0; iter < usersCount; iter++) {
+        await UsersFactory.Create({ name: `asdf_${faker.name.firstName()}` });
+      }
+    });
+    it('Should return an array', async () => {
+      const { result } = await UserModel.findWithSelect({},
+        'name, alias');
+      expect(Array.isArray(result)).to.be.true;
+    });
+    it('Should return correct user', async () => {
+      await UsersFactory.Create();
+      const { result } = await UserModel.findWithSelect({ name: { $in: [new RegExp('^asdf_')] } },
+        'name, alias');
+      expect(result.length).to.be.eq(usersCount);
+    });
+    it('Should check that the error exists', async () => {
+      const { error } = await UserModel.findWithSelect([]);
+      expect(error).to.be.exist;
+    });
+  });
+  describe('On getCustomCount', async () => {
+    let usersCount;
+    beforeEach(async () => {
+      await dropDatabase();
+      usersCount = _.random(1, 10);
+      for (let iter = 0; iter < usersCount; iter++) {
+        await UsersFactory.Create();
+      }
+    });
+    it('should return correct number', async () => {
+      const { count } = await UserModel.getCustomCount({});
+      expect(count).to.be.eq(usersCount);
+    });
+    it('Should return an error if condition transmitted incorrectly', async () => {
+      const { error } = await UserModel.getCustomCount(usersCount);
+      expect(error).to.be.exist;
+    });
+  });
```

Comment:
create conditions when not all users will be included in the sample, and check that the method correctly counted them